syntax = "proto3";

package transport_catalogue_serialize;

import "graph.proto";

message RouterSettings{
    int32 bus_wait_time = 1;
    double bus_velocity = 2;
}

message BusTripEdges{
    string bus_name = 1;
    double time = 2;
    uint32 span_count = 3;
    repeated string stops = 4;
};

message StopId{
    string stop = 1;
    uint32 id = 2;
}

message StopIdMap {
    repeated StopId stop_ids = 1;
}

message EdgeId{
    uint32 id = 1;
    BusTripEdges edge_info = 2;
}

message EdgeIdMap{
    repeated EdgeId edges_ids = 1;
}

message TransportRouter{
    RouterSettings router_settings = 1;
    StopIdMap stop_ids = 2;
    Graph graph = 3;
    EdgeIdMap edge_ids = 4;
}
